File: backend/revup/src/main/java/dev/michalzub/revup/reservation/ReservationDetailsModel.java
package dev.michalzub.revup.reservation;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@RequiredArgsConstructor
@Entity
@Table(name = "reservationDetails")
public class ReservationDetailsModel {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;
  private String customerName;
  private String customerSurname;
  private String customerPhone;
  private LocalDate startDate;
  private LocalDate endDate;
  private Integer totalPrice;

  @OneToOne
  @JoinColumn(name = "reservation_id")
  @JsonBackReference
  private ReservationModel reservation;
}

File: frontend/src/app/services/models/reservation-details-model.ts
/* tslint:disable */
/* eslint-disable */
import { ReservationModel } from '../models/reservation-model';
export interface ReservationDetailsModel {
  customerName?: string;
  customerPhone?: string;
  customerSurname?: string;
  endDate?: string;
  id?: number;
  reservation?: ReservationModel;
  startDate?: string;
  totalPrice?: number;
}
